### ISSUE TITLE
Most of my games not recognized.

### ISSUE THREAD

**Issue by TheStruggleForAntiSpaghetti** [RedDuckss/CemUI#45]
Hi,

Thanks for your work on this. I gave it a go and it only recognised Captain Toad Treasure Tracker from my games library folder.
I included a dirlist of the complete folder to enable you to investigate the reason.

Feel free to contact me if you need anything.

Thanks.

[dir.zip](https://github.com/RedDuckss/CemUI/files/1289935/dir.zip)

**Comment by RedDuckss** [comment]
The reason for this is because your roms folder structure is in a non-standard format. Your roms seem to all be RPX roms, which have a standard format of:
```
RomFolder
├── code
|   ├── // Code files
|   └── game.rpx
├── content
|   └── // Content files
└── meta
|   ├── // Meta files
|   └── meta.xml
```
Based on the list you gave, only Captain Toad follows this standard format. All your other games follow a non-standard format that uses the games product code/region-dev ID, like so:
```
RomFolder
└── ProductCodeOrRegionDevID // 'AMKP01' is a region-dev ID, 'WUP-P-ARPP' is a product code
|   ├── code
|   ├── content
|   └── meta
```

I chose to not use recursive lookups on purpose in order to prevent endless loops (in the event of a poorly made symlink) or long blocking (in the event of deeply rooted roms, like yours)

**Comment by RedDuckss** [comment]
The fact that these roms are not recognized is by design and is intended. However I will not close this issue so that discussion can continue unhindered and in case another user has the same problem, they can find the solution easier.

**Comment by sijedevos** [comment]
I have the same issue with super Mario 3D world and Mario kart 8. Both of these game have a folder with only one executable file (no code, content and meta folder) in it I believe .wud

**Comment by RedDuckss** [comment]
@sijedevos that is a separate issue, but related enough I guess. .wud and .wux roms cannot be in folders, they must be in the root. Take them out of the folder and you should be fine

**Comment by sijedevos** [comment]
@RedDuckss okay thanks I’ll give it a try

**Comment by sijedevos** [comment]
@RedDuckss it worked but I have one more problem. Super Luigi u didn’t seem to work either but this one did have the exact same structure listed above

**Comment by RedDuckss** [comment]
@sijedevos is it a .wud/.wux or an rpx rom

**Comment by sijedevos** [comment]
@RedDuckss it’s a rpx rom

**Comment by RedDuckss** [comment]
send me a zip of your meta folder

**Comment by sijedevos** [comment]
[meta.zip](https://github.com/RedDuckss/CemUI/files/1290958/meta.zip)

**Comment by RedDuckss** [comment]
@sijedevos your meta folder seems fine, so that wasnt the problem... This is no longer related to this issue though, please open a new issue report for this.

**Comment by sijedevos** [comment]
@RedDuckss okay thanks for the help anyway maybe redownloading the game works

**Comment by tuliocf3** [comment]
hello, i need help, i did all right but in the "Download Games" page don't pops up any game

**Comment by tuliocf3** [comment]
don't show anything

**Comment by Crinfarr** [comment]
tuliocf3 issues #100 #98 #99 #92 and #86 are all the same as your comment.  Please use one of those instead.

**Comment by timmehz** [comment]
I have the same problem with SMM (Super Mario Maker), it's not showing up in my library (I have FLUENT installed as my CemUI theme). I have updated my Cemu to 1.12.0d and I just got my CemUI just today, and that did not fix it. The file I want to get on my library is called Block.rpx here is some images of my folder setup.
![screenshot_1](https://user-images.githubusercontent.com/34901910/39789834-8aae230e-52ef-11e8-8724-325965a03a14.png)
![screenshot_2](https://user-images.githubusercontent.com/34901910/39789835-8ac92014-52ef-11e8-9d39-f5849424d57e.png)
![screenshot_3](https://user-images.githubusercontent.com/34901910/39789838-8af4a0e0-52ef-11e8-8f03-61ed7aedd553.png)

**Comment by RedDuckss** [comment]
@timmehz please read the comments already on this thread and read the flare. This is by design and won’t be fixed, since it isn’t a bug.

**Comment by timmehz** [comment]
I did but is there any way that I can fix this by myself?

**Comment by RedDuckss** [comment]
If you read the comments you would see the reason why this doesn’t work, which would tell you how to “fix” it (again, it doesn’t need “fixing”. It isn’t broken, this is intended).

To “fix” it you either need to put the folders in the correct structure or edit the source code yourself to allow deep searches.

**Comment by timmehz** [comment]
I read what you said again: 
RomFolder
└── ProductCodeOrRegionDevID // 'AMKP01' is a region-dev ID, 'WUP-P-ARPP' is a product code
|   ├── code
|   ├── content
|   └── meta

is ProductCodeOrRegionDevID a folder or a file? and if its a folder or a file what would it contain?

**Comment by RedDuckss** [comment]
Re-read what I said....I said that example is the _wrong_ format.

**Comment by RedDuckss** [comment]
Based on the images you sent before, no you do not.

When you selected the “Games” folder, all CemUI will look at is the “Super Mario Maker” folder.

It is _not_ looking inside the other folders, because it assumes “Super Mario Maker” is the rom folder with that structure, which it is not.

CemUI does not check the other folders because it does not implement any type of deep searching, which is intended and will not be changed.

**Comment by timmehz** [comment]
Well I have directed a folder destination in CemUI to check in where the Block.rpx file is located
![screenshot_4](https://user-images.githubusercontent.com/34901910/39790933-f71b089e-52f5-11e8-8225-509b3ef9d97e.png)

**Comment by RedDuckss** [comment]
That isn’t what you are supposed to do...as it told you during setup, when you select a “Games” folder it should be the folder that houses all of your roms....when CemUI checks to see if a folder is a rom folder, it checks for the “code”, “content” and “meta” folders...so when you try to point CemUI directly at the code folder, it’s going to check _in the code folder for the other folders_, which will fail....

**Comment by timmehz** [comment]
Um, I did what you said and this popped up when I try and direct it to the Games folder
![screenshot_5](https://user-images.githubusercontent.com/34901910/39791081-c0f6af10-52f6-11e8-872d-bcf0cf9dbf88.png)

**Comment by RedDuckss** [comment]
Did you remove all the wrong paths? There’s mostly likely a conflict going on with all the wrong paths you’ve created, or some corrupted config/storage file that was broken when making the wrong paths. I would suggest completely uninstalling and removing CemUI and then reinstalling it fresh

**Comment by timmehz** [comment]
ok

**Comment by timmehz** [comment]
and I might have removed the wrong paths because I did remove some of the directories to where it will find the games

**Comment by timmehz** [comment]
When I used the uninstaller to uninstall the CemUI and I also deleted the file that can install CemUI on your computer and I got a new copy, when I run the installer and when it tries to launch the CemUI it gives me the same error.

**Comment by RedDuckss** [comment]
I don't believe the uninstaller completely removes CemUI's data. When you uninstall it make sure that `%AppData%/CemUI` is removed as well. This is where CemUI stores it's data

**Comment by timmehz** [comment]
then if the uninstaller does not really uninstall all the data then should you not fix it to make it actually uninstall it? and yes, the CemUI folder is still there, I have deleted it just now.

**Comment by timmehz** [comment]
I have reinstalled it and I selected my Games folder and it gave me this error again:
![screenshot_6](https://user-images.githubusercontent.com/34901910/39791680-37465122-52fa-11e8-8aba-eda4f93e47a0.png)

**Comment by RedDuckss** [comment]
The uninstaller removes CemUI. That folder is not where CemUI is located. It is where CemUI stores it's other data like settings. This is common practice, as it means that non-program data is saved for if the user reinstalls the software (even major companies like Adobe do this).

Also, we did not make the installer/uninstaller. CemUI is compiled using a package called `electron-builder`, which to my knowledge does not even make an uninstaller (I am assuming you are using the built-in one in Windows, which we definitely cannot change).

```
I have reinstalled it and I selected my Games folder and it gave me this error again:
```
And is the `Games` folder still in the incorrect format? Because if so, then you've done nothing.

**Comment by timmehz** [comment]
Here is my structure on how I set up my Super Mario Maker folder
![screenshot_7](https://user-images.githubusercontent.com/34901910/39791813-d23fd630-52fa-11e8-80af-a6b2c1246d0a.png)

**Comment by RedDuckss** [comment]
Okay, good. That is the correct format.

The issue you have is now different than the one this topic is about. Please open a separate issue and let's move this discussion there, okay? Don't want to clutter up this thread with off-topic talk.

**Comment by timmehz** [comment]
ok

**Comment by Warbleshnarf** [comment]
When I downloaded a game found within the app while following a tutorial, I couldn't get the game to show up in the library ever.  I've read through this entire thread and nothing seems to work or even apply to what my folder looks like. Please help.

**Comment by RedDuckss** [comment]
> or even apply to what my folder looks like

Then create a new issue report or check the existing ones to see if an issue that does apply to you has already been reported. Please do not comment on issues that do not apply to your issue.
